### PR TITLE
fix: Prefer to break labels by '-' in UI

### PR DESCRIPTION
**Pull request by simster7**
Fixes https://github.com/argoproj/argo-workflows/issues/5020

Before: 
image

After:
image

### CODE REVIEW

File: ui/src/app/shared/components/graph/label.tsx [hunk 1]
```diff
@@ -9,6 +9,30 @@ export const formatLabel = (label: string) => {
         return <tspan>{label}</tspan>;
     }
     if (label.length <= maxPerLine * 2) {
+        const split = label.split('-');
+        if (split.length >= 2) {
```
Comment (alexec) [inline ui/src/app/shared/components/graph/label.tsx:13]: maybe we should have a unit test for this logic now?

Comment (alexec) [reply ui/src/app/shared/components/graph/label.tsx:13]: otherwise - nice work

Comment (simster7) [reply ui/src/app/shared/components/graph/label.tsx:13]: Absolutely, I'll work on some unit tests

Comment (simster7) [reply ui/src/app/shared/components/graph/label.tsx:13]: Tests done

File: ui/src/app/shared/components/graph/label.test.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import {formatLabel} from './label';
+
+const getLabels = (label: any): string[] => {
+    return [label.props.children[0].props.children, label.props.children[1].props.children];
+};
+
+describe('format label', () => {
```
Comment (alexec) [inline ui/src/app/shared/components/graph/label.test.ts:7]: nice!

**Review by alexec — APPROVED** [review]

**Pull request merged**